### ISSUE TITLE
[knockout] Distinguish between writable computeds and read-only computeds

### ISSUE THREAD

**Issue by Retsam** [DefinitelyTyped/DefinitelyTyped#18437]
A computed is only writable if it was provided a write function in it's arguments, but from the perspective of the current typings, all computed are writable.  It'd be nice if this code produced a Typescript error:

```
const randomNumber = ko.computed(() => 4) ;
randomNumber(6); //Will runtime error due to getRandomNumber being read-only
```

It seems feasible that the typings should be able to determine which computeds are writeable and disallow passing values to ones that aren't.  It seems like it'd be trivial to define a `KnockoutReadonlyComputed` type if Typescript supported [subtraction types](https://github.com/Microsoft/TypeScript/issues/4183), as it is, it seems possible but might require a bit of restructuring.

**Comment by hwanders** [comment]
This could also resolve an issue with `isWriteableObservable(instance: any): instance is KnockoutObservable<any>`: when it returns `false` TypeScript infers `instance` not being a `KnockoutObservable` at all, although it might be a readonly `KnockoutComputed`.

**Comment by ltlombardi** [comment]
@Retsam you implemented this some weeks ago. Could you close this issue?

**Comment by Retsam** [comment]
Ah, thanks for the reminder; I forgot that I opened this issue.

Fixed by #30499.
